packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart

```diff
@@ -442,13 +442,25 @@ class RetainingPathWidget extends StatelessWidget {
   /// parentMapKey, and parentField where applicable.
   String _retainingObjectDescription(RetainingObject object) {
     if (object.parentListIndex != null) {
-      final ref = object.value as InstanceRef;
-      return 'Retained by element [${object.parentListIndex}] of ${ref.classRef?.name ?? '<parentListName>'}';
+      final ref = object.value;
+      if (ref is InstanceRef) {
+        return 'Retained by element [${object.parentListIndex}] of'
+            ' ${ref.classRef?.name ?? '<parentListName>'}';
+      } else {
+        return 'Retained by element [${object.parentListIndex}] of '
```

Comment:
you can pull 'Retained by element [${object.parentListIndex}] of' into a local, as it is duplicated.

or just use a ternary inline:
```
return 'Retained by element [${object.parentListIndex}] of'
            '${ref is InstanceRef ?  ref.classRef?.name ?? '<parentListName>' :  _objectName(ref) ?? '<unknown>'}';
```

Reply:
I've made a helper for the container name `_containerClass`, and used it here and elsewhere